scipy/stats/_continuous_distns.py

```diff
@@ -5881,20 +5881,25 @@ def _rvs(self, df, size=None, random_state=None):
         return random_state.standard_t(df, size=size)
 
     def _pdf(self, x, df):
-        #                                gamma((df+1)/2)
-        # t.pdf(x, df) = ---------------------------------------------------
-        #                sqrt(pi*df) * gamma(df/2) * (1+x**2/df)**((df+1)/2)
-        r = np.asarray(df*1.0)
-        Px = (np.exp(sc.gammaln((r+1)/2)-sc.gammaln(r/2))
-              / (np.sqrt(r*np.pi)*(1+(x**2)/r)**((r+1)/2)))
-
-        return Px
+        return _lazywhere(
+            df == np.inf, (x, df),
+            f=lambda x, df: norm._pdf(x),
+            f2=lambda x, df: (
+                np.exp(sc.gammaln((df+1)/2)-sc.gammaln(df/2))
+                / (np.sqrt(df*np.pi)*(1+(x**2)/df)**((df+1)/2))
+            )
+        )
 
```

Comment:
Much better, thanks!
Do you think there is a numerical advantage to this separate implementation of `pdf` or would it be just as good or better to exponentiate `logpdf`?
It could be considered out of scope, but it's easier to review one function than two.
(That said, both of these look like they fix the issue and preserve behavior otherwise.)

Reply:
ping @tirthasheshpatel 
(I was thinking of merging this today)

Reply:
I am pretty happy with this so feel free to merge once tests pass!

Reply:
> Do you think there is a numerical advantage to this separate implementation of `pdf` or would it be just as good or better to exponentiate `logpdf`?

Sorry, I missed this! I think it's safe to exponentiate the `logpdf`. I don't see how the `pdf` implementation is more numerically stable than the `logpdf` version. On the contrary, `logpdf` should be more accurate on tails so exponentiation it should be better. I will change that and see if tests pass.

Reply:
The difference between the `t.pdf(x, df)` and `np.exp(t.logpdf(x, df))` is quite small. So, I don't think we need to worry too much about numerical stability:

```python
>>> import numpy as np
>>> from scipy.stats import t
>>> dist = t(5)
>>> x = np.linspace(0, dist.isf(1e-10), num=100_000)
>>> np.mean((dist.pdf(x) - np.exp(dist.logpdf(x)))**2)
3.5557301413708695e-35
>>> np.mean(np.abs(dist.pdf(x) - np.exp(dist.logpdf(x))))
5.971167081946323e-19
```

I can change the PDF to `np.log(self._logpdf(x, df))` in the PR only if it sounds good to you.